src/libstd/io/buffered.rs

```diff
@@ -300,6 +300,7 @@ impl<R: Seek> Seek for BufReader<R> {
 pub struct BufWriter<W: Write> {
     inner: Option<W>,
     buf: Vec<u8>,
+    panicked: bool,
```

Comment:
Could you add a comment explaining what this flag is? e.g. explaining where it's set, what it's used to detect, and what goes wrong if it's not here.